scheduler/service_cross_group_actions.go

```diff
@@ -0,0 +1,270 @@
+package scheduler
+
+import (
+	"fmt"
+	"github.com/Sirupsen/logrus"
+	"github.com/rancher/go-rancher-metadata/metadata"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const (
+	serviceLabel     = "io.rancher.stack_service.name"
+	zoneSplitsLabels = "io.rancher.scheduler.scale_per_group"
+)
+
+type ScaleGroupFilter struct {
+}
+
+func (s ScaleGroupFilter) Filter(scheduler *Scheduler, resourceRequest []ResourceRequest, context Context, hosts []string) []string {
+	if context == nil || len(context) == 0 || context[0].Data.Fields.Labels == nil {
+		return hosts
+	}
+	if serviceName, ok := context[0].Data.Fields.Labels[serviceLabel]; ok {
+		// service is not updated in metadata yet, update service manually
+		service := metadata.Service{}
+		if parts := strings.Split(serviceName, "/"); len(parts) == 2 {
+			service.StackName = parts[0]
+			service.Name = parts[1]
+			service.Labels = map[string]string{}
+			if val, ok := context[0].Data.Fields.Labels[zoneSplitsLabels]; ok {
```

Comment:
yes, it can be moved to the top level. We only care about service with that label